src/Analyser/NodeScopeResolver.php

```diff
@@ -4622,7 +4622,10 @@ private function processArgs(
 				$scope = $exprResult->getScope();
 				$hasYield = $hasYield || $exprResult->hasYield();
 
-				if ($exprType->isCallable()->yes()) {
+				if (
+					$exprType->isCallable()->yes()
+					&& (!$exprType instanceof NeverType || $exprType->isExplicit())
```

Comment:
a different way to fix this would be to make `NeverType` return a different acceptor on `getCallableParametersAcceptors`

Reply:
Yeah, that would make more sense to me. It's basically never going to be called so a special class instead of TrivialParametersAcceptor would be welcome. It's pure, and probably also returns NeverType?